test/lightning_web/live/job_live/manual_run_component_test.exs

```diff
@@ -283,7 +291,43 @@ defmodule LightningWeb.JobLive.ManualRunComponentTest do
              job: job,
              current_user: user,
              on_run: nil,
-             builder_state: %{job_id: job.id, dataclip: d4}
+             builder_state: %{job_id: job.id, dataclip: d4},
+             can_edit_job: true,
+             return_to:
+               Routes.project_workflow_path(
+                 conn,
+                 :show,
+                 project.id,
+                 job.workflow_id
+               )
            ) =~ "<option selected value=\"#{d4.id}\">#{d4.id}</option>"
   end
+
+  test "project viewers can't run a job from the inspector", %{
+    conn: conn,
+    project: project
+  } do
+    {conn, _user} = setup_project_user(conn, project, :viewer)
+
+    job =
+      workflow_job_fixture(
+        project_id: project.id,
+        body: ~s[fn(state => { return {...state, extra: "data"} })]
+      )
+
+    {:ok, view, _html} =
+      live(
+        conn,
+        ~p"/projects/#{project.id}/w/#{job.workflow_id}/j/#{job.id}"
+      )
+
+    assert view
+           |> element("button[phx-click='confirm'][disabled]")
+           |> has_element?()
+
+    view
+    |> with_target("#manual-job-#{job.id}")
+    |> render_click("confirm", %{}) =~
+      "You are not authorized to perform this action."
```

Comment:
This was the approach I mentioned last time we were pairing, the way to use it is to target the _live component_ not the element with the `phx-click` (in this specific case).

Reply:
Thanks a lot for this one